src/main/java/net/sf/jabref/importer/fileformat/BibTeXMLImporter.java

```diff
@@ -78,33 +102,539 @@ public ParserResult importDatabase(BufferedReader reader) throws IOException {
+                    parseInproceedings(entry.getInproceedings(), fields);
+                }
+                if (entry.getManual() != null) {
+                    bibEntry.setType("manual");
+                    parseManual(entry.getManual(), fields);
+                }
+                if (entry.getMastersthesis() != null) {
+                    bibEntry.setType("mastersthesis");
+                    parseMasterthesis(entry.getMastersthesis(), fields);
+                }
+                if (entry.getMisc() != null) {
+                    bibEntry.setType("misc");
+                    parseMisc(entry.getMisc(), fields);
+                }
+                if (entry.getPhdthesis() != null) {
+                    bibEntry.setType("phdthesis");
+                    parsePhdthesis(entry.getPhdthesis(), fields);
+                }
+                if (entry.getProceedings() != null) {
+                    bibEntry.setType("proceedings");
+                    parseProceedings(entry.getProceedings(), fields);
+                }
+                if (entry.getTechreport() != null) {
+                    bibEntry.setType("techreport");
+                    parseTechReport(entry.getTechreport(), fields);
+                }
+                if (entry.getUnpublished() != null) {
+                    bibEntry.setType("unpublished");
+                    parseUnpublished(entry.getUnpublished(), fields);
+                }
+                if (entry.getId() != null) {
+                    bibEntry.setCiteKey(entry.getId());
+                }
+                bibEntry.setField(fields);
+                bibItems.add(bibEntry);
+            }
+        } catch (JAXBException | XMLStreamException e) {
             LOGGER.error("Error with XML parser configuration", e);
             return ParserResult.fromErrorMessage(e.getLocalizedMessage());
-        } catch (org.xml.sax.SAXException e) {
-            LOGGER.error("Error during XML parsing", e);
-            return ParserResult.fromErrorMessage(e.getLocalizedMessage());
-        } catch (IOException e) {
-            LOGGER.error("Error during file import", e);
-            return ParserResult.fromErrorMessage(e.getLocalizedMessage());
         }
         return new ParserResult(bibItems);
     }
 
+    private void parseUnpublished(Unpublished unpublished, HashMap<String, String> fields) {
+        putIfValueNotNull(fields, FieldName.ABSTRACT, unpublished.getAbstract());
+        putIfValueNotNull(fields, "affiliation", unpublished.getAffiliation());
+        putIfValueNotNull(fields, FieldName.AUTHOR, unpublished.getAuthor());
+        putIfValueNotNull(fields, "category", unpublished.getCategory());
+        putIfValueNotNull(fields, "copyright", unpublished.getCopyright());
+        putIfValueNotNull(fields, FieldName.CROSSREF, unpublished.getCrossref());
+        putIfValueNotNull(fields, FieldName.ISBN, unpublished.getIsbn());
+        putIfValueNotNull(fields, FieldName.ISSN, unpublished.getIssn());
```

Comment:
According to the bibtexml.xsd file an entry can have either an isbn or an issn number but not both.